.expeditor/scripts/finish_release/bump_rustfmt.sh

```diff
@@ -0,0 +1,92 @@
+#!/bin/bash
+
+# This script attempts to find the latest installable nightly rustfmt.
+# When found, it bumps the repo rustfmt version and submits a new PR
+# with the new version pin and any code base formatting changes from the
+# new version.
+
+# Assumptions:
+# 1. This script runs on a x86_64-unknown-linux-gnu target
+# 2. If rustfmt is installable on x86_64-unknown-linux-gnu, the other Tier1
+#    platforms are assumed to also have a working version.
+
+set -Eeuo pipefail
+
+# shellcheck source=.expeditor/scripts/verify/shared.sh
+source .expeditor/scripts/verify/shared.sh
+
+TODAY=$(date '+%Y-%m-%d')
+# pick out the date from a string like: nightly-2019-12-04
+CURRENT_RUSTFMT_DATE=$(awk -F- '{print $2"-"$3"-"$4}' RUSTFMT_VERSION)
+# convert to seconds for easier date comparison later
+CURRENT_RUSTFMT_DATE_S=$(date '+%s' --date "${CURRENT_RUSTFMT_DATE}")
+CHANNEL=nightly
+# a maximum iterator value for searching elligible nightlies
+MAX_ATTEMPTS=90
+
+# if set, this will be the next version
+export FOUND_VERSION
+
+install_rustup
+
+# set minimal profile so only core items + rustfmt are installed
+echo "--- :habicat: Setting minimal profile"
+rustup set profile minimal
+
+for ((i=0; i<MAX_ATTEMPTS; i++)); do
```

Comment:
It sounds like `rustup` can handle a lot of this logic automatically: https://blog.rust-lang.org/2019/10/15/Rustup-1.20.0.html#installing-the-latest-compatible-nightly.

Would that help simplify this?

Reply:
The trouble there is that logic is designed AFAICT for situations where you do e.g.
```shell
rustup install nightly
rustup update
```
Which doesn't really apply to us.  Since we always install a version of nightly with an explicit date, `rustup update` doesn't make any sense.

Reply:
Yes, I agree, it's not as useful as I had originally hoped 🙁 

To illustrate, I uninstalled all my toolchains, installed our previous version (nightly-2019-10-15) and then ran `rustup update` and it didn't report any available updates even though I followed up with a manual install of our current version (nightly-2019-11-14) successfully. 

```
✔ ~/Projects/habitat [jeremymv2/auto_bump_rustfmt L|…1⚑ 4]
15:23 $ rustup toolchain list
nightly-2019-10-15-x86_64-apple-darwin
1.39.0-x86_64-apple-darwin

✔ ~/Projects/habitat [jeremymv2/auto_bump_rustfmt L|…1⚑ 4]
15:23 $ rustup update
info: no updatable toolchains installed
info: checking for self-updates

✔ ~/Projects/habitat [jeremymv2/auto_bump_rustfmt L|…1⚑ 4]
15:23 $ rustup toolchain install $(<RUSTFMT_VERSION)
info: syncing channel updates for 'nightly-2019-11-14-x86_64-apple-darwin'
info: latest update on 2019-11-14, rust version 1.41.0-nightly (ded5ee001 2019-11-13)
info: downloading component 'cargo'
info: downloading component 'clippy'
info: downloading component 'rust-docs'
info: downloading component 'rust-std'
info: downloading component 'rustc'
....

  nightly-2019-11-14-x86_64-apple-darwin installed - rustc 1.41.0-nightly (ded5ee001 2019-11-13)

info: checking for self-updates
✔ ~/Projects/habitat [jeremymv2/auto_bump_rustfmt L|…1⚑ 4]
15:24 $
```